pkg/services/baremetal/host/host.go

```diff
@@ -592,46 +592,18 @@ func (s *Service) actionRegistering(_ context.Context) actionResult {
 	}
 	sshClient := s.scope.SSHClientFactory.NewClient(in)
 
-	// Check hostname with sshClient
-	out := sshClient.GetHostName()
-	hostName := trimLineBreak(out.StdOut)
-	if hostName != rescue {
-		// give the reboot some time until it takes effect
-		if s.hasJustRebooted() {
-			return actionContinue{delay: 2 * time.Second}
-		}
-
-		isSSHTimeoutError, isSSHConnectionRefusedError, err := s.analyzeSSHOutputRegistering(out)
-		if err != nil {
-			// This can happen if the bare-metal server was taken by another mgt-cluster.
-			// Check in https://robot.hetzner.com/server for the "History" of the server.
-			return actionError{err: fmt.Errorf("failed to handle incomplete boot - registering: %w", err)}
-		}
-
-		failed, err := s.handleIncompleteBoot(true, isSSHTimeoutError, isSSHConnectionRefusedError)
-		if failed {
-			return s.recordActionFailure(infrav1.PermanentError, err.Error())
-		}
-		if err != nil {
-			return actionError{err: fmt.Errorf(errMsgFailedHandlingIncompleteBoot, err)}
-		}
-		timeSinceReboot := "unknown"
-		if s.scope.HetznerBareMetalHost.Spec.Status.LastUpdated != nil {
-			timeSinceReboot = time.Since(s.scope.HetznerBareMetalHost.Spec.Status.LastUpdated.Time).String()
-		}
-
-		s.scope.Logger.Info("Could not reach rescue system. Will retry some seconds later.", "out", out.String(), "hostName", hostName,
-			"isSSHTimeoutError", isSSHTimeoutError, "isSSHConnectionRefusedError", isSSHConnectionRefusedError, "timeSinceReboot", timeSinceReboot)
-		return actionContinue{delay: 10 * time.Second}
+	ok, res := s.validateRescueSystemIsActive(sshClient)
+	if !ok {
+		return res
 	}
-
 	output := sshClient.GetHardwareDetailsDebug()
 	if output.Err != nil {
 		return actionError{err: fmt.Errorf("failed to obtain hardware for debugging: %w", output.Err)}
 	}
+
 	msg := fmt.Sprintf("%s\n\n", output.StdOut)
-	if out.StdErr != "" {
-		msg += fmt.Sprintf("stderr:\n%s\n\n", out.StdErr)
+	if output.StdErr != "" {
```

Comment:
This was a typo in the old code, which as revealed by extracting the new method.